bool save_unknown
---
bool success
